module/VuFind/src/VuFind/Marc/MarcReader.php

```diff
@@ -277,4 +277,161 @@ public function getFieldsSubfields(string $fieldTag, array $subfieldCodes,
+                    'data' => current($subfield),
+                ];
+            }
+            if ($subfields) {
+                $result[] = [
+                    'tag' => $fieldTag,
+                    'i1' => $field['i1'],
+                    'i2' => $field['i2'],
+                    'subfields' => $subfields,
+                    'link' => $link
+                ];
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Return an array of all values extracted from the specified linked
+     * field/subfield combination.  If multiple subfields are specified and $concat
+     * is true, they will be concatenated together in the order listed -- each entry
+     * in the array  will correspond with a single MARC field.  If $concat is false,
+     * the return array will contain separate entries for separate subfields.
+     *
+     * @param string $fieldTag       The MARC field that contains the linked fields
+     * @param string $linkedFieldTag The linked MARC field tag to get
+     * @param array  $subfieldCodes  The MARC subfield codes to get
+     * @param bool   $concat         Should we concatenate subfields?
+     * @param string $separator      Separator string (used only if $concat === true)
+     *
+     * @return array
+     */
+    public function getLinkedFieldsSubfields(string $fieldTag,
+        string $linkedFieldTag, array $subfieldCodes, bool $concat = true,
+        string $separator = ' '
+    ): array {
+        $result = [];
+        foreach ($this->getLinkedFields($fieldTag, $linkedFieldTag, $subfieldCodes)
+            as $field
+        ) {
+            $subfields = $this->getSubfields($field);
+            $result[] = $concat ? implode($separator, $subfields) : $subfields;
+        }
+        return $result;
+    }
+
+    /**
+     * Get linked field data from subfield 6
+     *
+     * @param array $field Field
+     *
+     * @return array
+     */
+    public function getFieldLink(array $field): array
+    {
+        return $this->parseLinkageField($this->getSubfield($field, '6'));
+    }
+
+    /**
+     * Parse a linkage field
+     *
+     * @param string $link Linkage field
+     *
+     * @return array
+     */
+    public function parseLinkageField(string $link): array
+    {
+        $linkParts = explode('/', $link, 3);
+        $targetParts = explode('-', $linkParts[0]);
+        return [
+            'field' => $targetParts[0],
+            'occurrence' => $targetParts[1] ?? '',
+            'script' => $linkParts[1] ?? '',
+            'orientation' => $linkParts[2] ?? ''
+        ];
+    }
+
+    /**
+     * Return first subfield with the given code in the internal MARC field
+     *
+     * @param array  $field        Result from MarcReader::getFields
+     * @param string $subfieldCode The MARC subfield code to get
+     *
+     * @return string
+     */
+    protected function getInternalSubfield(array $field, string $subfieldCode
```

Comment:
This function confused me a little bit. I think I understand that this is for working with the internal, abbreviated array, which differs from the more verbose public interface exposed by the class... but the documentation for the $field parameter is identical for this function and for the public `getSubfield()` version. Is one of them incorrect?